File: Elmah.Contrib.EntityFramework/EntityErrorLog.fs
namespace Elmah.Contrib.EntityFramework

open System.Data.Entity.Core.EntityClient

open Elmah

[<Sealed>]
type public EntityErrorLog(config : System.Collections.IDictionary) = 
    inherit SqlErrorLog(config)

    override this.ConnectionString
        with get() =
            let builder = new EntityConnectionStringBuilder(base.ConnectionString)
            builder.ProviderConnectionString

File: Elmah.Contrib.EntityFramework/EntityErrorLog.cs
using System.Data.Entity.Core.EntityClient;

namespace Elmah.Contrib.EntityFramework
{
	public class EntityErrorLog : SqlErrorLog
	{
		public EntityErrorLog(string connectionString)
			: base(connectionString)
		{
		}

		public EntityErrorLog(System.Collections.IDictionary config)
			: base(config)
		{
		}

		public override string ConnectionString
		{
			get
			{
				var builder = new EntityConnectionStringBuilder(base.ConnectionString);
				return builder.ProviderConnectionString;
			}
		}
	}
}